Automattic/pocket-casts-android
# Track application lifecycle on watch

## Description
This PR enables the application lifecycle tracking of the watch app (issue https://github.com/Automattic/pocket-casts-android/issues/1007).

As part of this PR, I have added a "shared" module for code that is shared between the automotive, phone, and wear apps. I've had a few times come up where I thought this would be helpful, so I think it will be worthwhile even apart from the changes in this PR, but let me know if you disagree.

## Testing Instructions
1. Install the watch app
2. Launch the watch app
3. ✅  Observe the `application_installed` and `application_opened` events are fired
4. Background the wear app
5. ✅ Observe the `application_closed` event is fired

## Checklist
 
#### I have tested any UI changes...
